agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java

```diff
@@ -352,8 +353,12 @@ public static void logConfigurationWarnMessages() {
 
   // visible for testing
   static void overlayFromEnv(Configuration config) throws IOException {
-    config.connectionString = overlayConnectionStringFromEnv(config.connectionString);
-
+    // load connection string from a file if connection string is in the format of
+    // "${file:mounted_connection_string_file.txt}"
+    StringSubstitutor stringSubstitutor = new StringSubstitutor(FileStringLookup.INSTANCE);
+    stringSubstitutor.setEnableSubstitutionInVariables(true);
```

Comment:
is this needed?

Reply:
yes, otherwise, the result will return "[value]".